Lib/ldap/controls/readentry.py

```diff
@@ -42,7 +42,7 @@ def decodeControlValue(self,encodedControlValue):
     self.dn = str(decodedEntry[0])
     self.entry = {}
     for attr in decodedEntry[1]:
-      self.entry[str(attr[0])] = [ str(attr_value) for attr_value in attr[1] ]
+      self.entry[str(attr[0])] = [ bytes(attr_value) for attr_value in attr[1] ]
```

Comment:
Is ``bytes()`` required at all? The attributes should be bytes already.

A simple test case and updated documentation would be fantastic.

Reply:
Yes, bytes is required. As can be seen in #399 it's a `<AttributeValue value object at 0x7efe18dbe518 tagSet <TagSet object at 0x7efe1b0569e8 tags 0:0:4> encoding iso-8859-1 payload [0xc384646d696e6973747261746f7234]>`

Reply:
Test case has been added.

Reply:
There are no docs.